packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -60,82 +44,193 @@ export interface TokenAuthorizerProps {
+    }
+
+    this.restApiId = restApi.restApiId;
+  }
+
+  /**
+   * Sets up the permissions necessary for the API Gateway service to invoke the Lambda function.
+   */
+  protected setupPermissions() {
+    if (!this.role) {
+      this.handler.addPermission(`${this.node.uniqueId}:Permissions`, {
+        principal: new iam.ServicePrincipal('apigateway.amazonaws.com'),
+        sourceArn: this.authorizerArn
+      });
+    } else if (this.role instanceof iam.Role) { // i.e. not imported
+      this.role.attachInlinePolicy(new iam.Policy(this, 'authorizerInvokePolicy', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: [ this.handler.functionArn ],
+            actions: [ 'lambda:InvokeFunction' ],
+          })
+        ]
+      }));
+    }
+  }
+}
+
+/**
+ * Properties for TokenAuthorizer
+ */
+export interface TokenAuthorizerProps extends LambdaAuthorizerProps {
+  /**
+   * An optional regex to be matched against the authorization token. When matched the authorizer lambda is invoked,
+   * otherwise a 401 Unauthorized is returned to the client.
+   *
+   * @default - no regex filter will be applied.
+   */
+  readonly validationRegex?: string;
 
+  /**
+   * The request header mapping expression for the bearer token. This is typically passed as part of the header, in which case
+   * this should be `method.request.header.Authorizer` where Authorizer is the header containing the bearer token.
+   * @see https://docs.aws.amazon.com/apigateway/api-reference/link-relation/authorizer-create/#identitySource
+   * @default `IdentitySource.header('Authorization')`
+   */
+  readonly identitySource?: string;
+}
+
+/**
+ * Token based lambda authorizer that recognizes the caller's identity as a bearer token,
+ * such as a JSON Web Token (JWT) or an OAuth token.
+ * Based on the token, authorization is performed by a lambda function.
+ *
+ * @resource AWS::ApiGateway::Authorizer
+ */
+export class TokenAuthorizer extends LambdaAuthorizer {
+
+  public readonly authorizerId: string;
+
+  public readonly authorizerArn: string;
+
+  constructor(scope: Construct, id: string, props: TokenAuthorizerProps) {
+    super(scope, id, props);
+
+    const restApiId = Lazy.stringValue({ produce: () => this.restApiId });
     const resource = new CfnAuthorizer(this, 'Resource', {
       name: props.authorizerName ?? this.node.uniqueId,
       restApiId,
       type: 'TOKEN',
       authorizerUri: `arn:aws:apigateway:${Stack.of(this).region}:lambda:path/2015-03-31/functions/${props.handler.functionArn}/invocations`,
-      authorizerCredentials: props.assumeRole ? props.assumeRole.roleArn : undefined,
-      authorizerResultTtlInSeconds: props.resultsCacheTtl && props.resultsCacheTtl.toSeconds(),
+      authorizerCredentials: props.assumeRole?.roleArn,
+      authorizerResultTtlInSeconds: props.resultsCacheTtl?.toSeconds() ?? Duration.minutes(5).toSeconds(),
```

Comment:
(non-blocking)

You shouldn't need this. It's common in the CDK to leave this undefined and document the CF's default as part of the `@default` annotation in the tsdoc.

Reply:
This is needed to fix #2561, but we can get rid of it and just wait for Cfn to issue a fix?

Reply:
I'm ok with you leaving this in if you prefer.

However, can you describe how this fixes #2561?

Reply:
I think the issue is that the previous iteration would evaluate to a boolean (`false`) which got converted into an explicit `0`, which would override the default 5 minutes. This evaluates it to `undefined`, which maintains the default. So this still fixes the issue, and also doesn't explicitly define the default, which is what we want. (Could probably also close the internal issue)

Reply:
I'm still not following this entirely. What would evaluate to a boolean (`false`)?

AFAICT, in the previous iteration, if the user does not specify a `resultsCacheTtl`, it's set to `undefined` in the CloudFormation resource. This (correctly) evaluates to the property not being set in the CF template, and hence let CF use its default.

It's possible that I've misunderstood your comment. If so, feel free to correct me.

Reply:
Ok yes, my confusion came from the fact that this was originally failing integ tests for the same reason as specified in that issue, and with them now passing here that issue could likely be closed. But I've re-read it, and their issue stems from required/optional properties in Cfn (as you probably know).

So this doesn't fix #2561, but the new syntax is a little cleaner and is identical in functionality to the original iteration.